firedrake/mesh.py

```diff
@@ -1936,6 +1971,8 @@ def Mesh(meshfile, **kwargs):
     geometric_dim = kwargs.get("dim", None)
     if isinstance(meshfile, PETSc.DMPlex):
         plex = meshfile
+    elif netgen and isinstance(meshfile, netgen.libngpy._meshing.Mesh):
+        plex = _from_netgen(meshfile, comm)
```

Comment:
What do you mean for name ?

Reply:
Name of the mesh (such as `my_mesh_A` or `cylinder_mesh_B`), which we need in order to save the mesh in file at later stage if needed. When we call this `Mesh` function, `name` is passed as an argument (otherwise the default value is assigned to `name`); I thought the Netgen mesh object would carry some name with it, but it seemed not.

Reply:
Ah, I'll have a look into it. Maybe it is possible to add this label on the Netgen side.

Reply:
I've added this feature in my Netgen fork, now there are a `SetMeshName(name)` and `GetMeshName(name)` functions.